src/hotspot/share/runtime/arguments.cpp

```diff
@@ -3972,6 +3972,19 @@ jint Arguments::parse(const JavaVMInitArgs* initial_cmd_args) {
   no_shared_spaces("CDS Disabled");
 #endif // INCLUDE_CDS
 
+#ifdef LEAK_SANITIZER
+#ifdef _LP64
+  // LSAN relies on pointers to be natively aligned. Using compressed class pointers breaks this
+  // expectation and results in nondeterministic leak reports.
+  if (FLAG_SET_CMDLINE(UseCompressedOops, false) != JVMFlag::SUCCESS) {
```

Comment:
This positioning doesn't look right. There are a number of inter-dependencies on the compressed oops flags and I don't think you can just turn them off here. Surely `Arguments::set_use_compressed_oops()` is where this needs to be done?

Reply:
Not sure FLAG_SET_CMDLINE is right here either (but flag setting is very confusing).

Reply:
Removed, no longer needed. I'll explain below.